src/utils/hooks/api/oracle/use-get-oracle-status.ts

```diff
@@ -0,0 +1,31 @@
+import { useQuery } from 'react-query';
+
+import { REFETCH_INTERVAL } from '@/utils/constants/api';
+
+interface UseGetOracleStatusResult {
+  data: OracleStatus | undefined;
+  isLoading: boolean;
+}
+
+enum OracleStatus {
+  ONLINE = 'ONLINE',
+  OFFLINE = 'OFFLINE'
+}
+
+const getOracleStatus = async (): Promise<OracleStatus> => {
+  const isOracleOnline = await window.bridge.oracle.isOnline();
+  return isOracleOnline ? OracleStatus.ONLINE : OracleStatus.OFFLINE;
+};
+
+const useGetOracleStatus = (): UseGetOracleStatusResult => {
+  const { data, isLoading } = useQuery({
+    queryKey: 'oracle-status',
+    queryFn: getOracleStatus,
+    enabled: window.bridge !== undefined,
+    refetchInterval: REFETCH_INTERVAL.MINUTE
+  });
+
```

Comment:
Should we check for  `useErrorHandler(error)` in case the call fails?

Reply:
Don't think that's needed in this case - the call fails if RPC or client is offline in which case the status should be displayed as offline :)